sdk/dapp-kit/src/components/styling/InjectedThemeStyle.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+
+import { styleDataAttributeSelector } from '../../constants/styleDataAttribute.js';
+import { themeVars } from '../../themes/themeContract.js';
+import type { Theme, ThemeVars } from '../../themes/themeContract.js';
+
+type InjectedThemeStyleProps = {
+	theme: Theme;
+};
+
+export function InjectedThemeStyle({ theme }: InjectedThemeStyleProps) {
+	const defaultStyles = `${styleDataAttributeSelector}{${cssStringFromTheme(
```

Comment:
that looks weird. Is this the new way to write css? The wrapping parens are going to be annoying to write.

Reply:
this crazy component injects the theme definitions into the DOM tree that are properly scoped to only `dapp-kit` components 😄. It is inlined like this (A) for supporting SSR and (B) not having to be concerned with individually injecting each theme variable into UI components

Reply:
@plam-ml Everything in dapp-kit is different than our normal products because we have a lot more limitations with shipping components, so don't stress out too much about this.